pkg/controllermanager/controller/project/project_control_reconcile.go

```diff
@@ -48,6 +48,7 @@ func (r *projectReconciler) reconcile(ctx context.Context, log logr.Logger, proj
 		if err := controllerutils.StrategicMergePatchAddFinalizers(ctx, gardenClient, project, gardencorev1beta1.GardenerName); err != nil {
 			return reconcile.Result{}, fmt.Errorf("could not add finalizer to Project: %w", err)
 		}
+		return reconcile.Result{}, nil
```

Comment:
I'm not sure if this will be enough to prevent multiple reconciliations for the project. I see that the `project.Status.ObservedGeneration` is updated only at the end of the reconciliation, which means that any other updates to the resource would requeue it since: https://github.com/gardener/gardener/blob/710ca095fa8c88fd9bf3589f2879beb236e2610d/pkg/controllermanager/controller/project/project_control.go#L66-L70

 will always be true

Reply:
As this PR only deals with reconciliations caused by finalizer addition, will it be better to track the above mentioned change in a different issue. WDYT?